src/components/MovieListPage/movieListPage.jsx

```diff
@@ -23,17 +24,47 @@ const MovieListPage = (props) => {
     const [isLoading, setIsLoading] = useState(false);
     const [movies, setMovies] = useState([]); //movie list
 
+    const navigate = useNavigate();
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    useEffect(() => {
+        setSearchParams({ sortCriterion, searchQuery, activeGenre });
+    }, [sortCriterion, searchQuery, activeGenre]);
+
+    useEffect(() => {
+        if(movieId) navigate(`/${movieId}`);
+    }, [movieId]);
+
+    useEffect(() => {
+      const params = new URLSearchParams(searchParams);
+      const newSortCriterion = params.get('sortCriterion') || 'release_date';
+      const newSearchQuery = params.get('searchQuery') || searchQuery;
+      const newActiveGenre = params.get('activeGenre') || 'All';
+      setSortCriterion(newSortCriterion);
+      setActiveGenre(newActiveGenre);
+      setSearchQuery(newSearchQuery);
+    }, [searchParams]);
+
     useEffect(() => {
         setIsLoading(true);
-        const query = buildQuery();
-        fetch(`http://localhost:4000/movies?${query}`)
-          .then(response => response.json())
-          .then(data => {
+        if(showDetail && movieId) {
+            fetch(`http://localhost:4000/movies/${movieId}`)
+            .then(response => response.json())
+            .then(data => {
+                setSelectedMovie(data[0]);
+                setIsLoading(false);
+            })
+            .catch(error => console.error(error));
+        }
+
+        fetch(`http://localhost:4000/movies?${buildQuery()}`)
+        .then(response => response.json())
+        .then(data => {
             setMovies(data.data);
```

Comment:
then(({data}) => {
            setMovies(data);
//etc

Reply:
When I use this fetch, I am getting an object as a response. In this object, I have "data", "limit", "offset", and "totalAmount" elements.